### PR TITLE
Documentation of BinaryCodecs

### PR DESCRIPTION
**Pull request by khajavi**

### CODE REVIEW

**Comment by khajavi** [comment]
Before merging this PR, please merge this one: #2785

**Comment by khajavi** [comment]
@987Nabil 
Do you have any idea why the CI failed on this PR?

File: zio-http-example/src/main/scala/example/codecs/RequestBodyJsonDeserializationExample.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  val app: Routes[Ref[List[Book]], Nothing] =
+    Routes(
+      Method.POST / "books" ->
+        handler { (req: Request) =>
+          for {
+            book  <- req.body.to[Book].catchAll(_ => ZIO.fail(Response.badRequest("unable to deserialize the request")))
+            books <- ZIO.service[Ref[List[Book]]]
+            _     <- books.updateAndGet(_.appended(book))
```
Comment (987Nabil) [inline zio-http-example/src/main/scala/example/codecs/RequestBodyJsonDeserializationExample.scala]: try instead `_  :+ book`

Comment (khajavi) [reply zio-http-example/src/main/scala/example/codecs/RequestBodyJsonDeserializationExample.scala]: Thank you! Fixed!
I'm curious, what is the difference between `:+` and `appended`?

**Review by jdegoes — APPROVED** [review]

**Pull request merged**